docs/getting_started.rst

```diff
@@ -1,135 +1,288 @@
-Now that we have our data and the storage location, we can persist it to a dataset.
-For that we use in this guide :func:`kartothek.io.eager.store_dataframes_as_dataset`
-to store a ``DataFrame`` we already have in memory in the local task.
+For a more technical description of storage specification in kartothek, see
+:ref:`storage_specification`.
 
-The import path of this function already gives us a hint about the general
-structuring of the ``kartothek`` modules. In :mod:`kartothek.io` we have all
-the building blocks to build data pipelines that read and write from/to storages.
-Other top-level modules for example handle the serialization of DataFrames to
-``bytes``.
 
-The next module level ``eager`` describes the scheduling backend.
-``eager`` runs all execution immediately and on the local machine.
-There is also ``iter`` that supports reading the dataset on
-a per-partition base. For larger dataset, ``dask`` can be used
-to work on datasets in parallel or even in a cluster by using
-``distributed`` as the backend for ``dask``.
+Writing data to storage
+=======================
+
+Now that we have some data and a location to store it in, we can persist it as a
+dataset. To do so, we will use :func:`~kartothek.io.eager.store_dataframes_as_dataset`
+to store the ``DataFrame`` ``df`` that we already have in memory.
 
 .. ipython:: python
-   :okwarning:
 
-   from kartothek.io.eager import store_dataframes_as_dataset
-   dm = store_dataframes_as_dataset(
-      store, "a_unique_dataset_identifier", df, metadata_version=4
-   )
-   dm
+    from kartothek.io.eager import store_dataframes_as_dataset
+
+    dm = store_dataframes_as_dataset(store_factory, "a_unique_dataset_identifier", df)
+    dm
 
-After calling :func:`~kartothek.io.eager.store_dataframes_as_dataset`,
-a :class:`kartothek.core.dataset.DatasetMetadata` object is returned. This is the main
-class holding all information about the parts and schema of the dataset.
 
-The most interesting ones for this guide are ``tables`` and ``partitions``.
-Each dataset can have multiple tables, each table is a collection of files
-that all have the same schema. These files are called ``partitions`` in
-``kartothek``.
+.. admonition:: Storage backends
 
-As we neither have explicitly defined the name of the table nor the name
-of the created partition, ``kartothek`` has used the default table name
-``table`` and used a generated UUID for the partition name.
+    The import path of this function already gives us a hint about the general
+    structuring of the ``kartothek`` modules. In :mod:`kartothek.io` we have all
+    the building blocks to build data pipelines that read and write from/to storages.
+    The next module level (e.g. ``eager``) describes the scheduling backend.
+
+    The scheduling backends `currently supported` by kartothek are:
+
+    - ``eager`` runs all execution immediately and on the local machine.
+    - ``iter`` executes operations on the dataset on a per-partition basis.
```

Comment:
The per-partition bit is true for all other backends as well. The important part is that it is a python generator/iterator interface.